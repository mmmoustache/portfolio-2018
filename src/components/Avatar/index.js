import styled from "styled-components";
import { Breakpoints } from '../../utils/breakpoints';
import { Timings, Durations } from '../../utils/transitions';
import { convertRemToPixels as rem } from '../../utils/utils';

const size = {
  small: `${rem('200px')}`,
  large: `${rem('300px')}`,
};

export default styled.div`
  background-image: url("./me.jpg");
  background-position: center;
  background-size: 110%;
  clip-path: circle(0% at 50% 50%);
  display: block;
  height: ${size.small};
  margin: 0 auto ${rem('60px')};
  overflow: hidden;
  width: ${size.small};

  @media ${Breakpoints.medium} {
    height: ${size.large};
    margin-bottom: 0;
    width: ${size.large};
  }

  ${
    props => props.isVisible && `
      animation-name: avatar;
      animation-fill-mode: forwards;
      animation-duration: ${Durations.default};
      animation-timing-function: ${Timings.default};
    `
  }
  
  @keyframes avatar {
    0% {
      clip-path: circle(0% at 50% 50%);
    }
    100% {
      clip-path: circle(50% at 50% 50%);
    }
  }
`;
