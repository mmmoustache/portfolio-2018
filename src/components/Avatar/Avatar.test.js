import React from 'react';
import renderer from 'react-test-renderer';
import Avatar from '../Avatar';

describe('Component: Avatar', () => {

  it('Should render successfully', () => {
    const component = renderer.create(
      <Avatar />
    );

    expect(component.toJSON()).toMatchSnapshot();
  });

});
