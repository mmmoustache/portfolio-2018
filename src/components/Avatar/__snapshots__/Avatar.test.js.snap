// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Avatar Should render successfully 1`] = `
<div
  className="sc-bdVaJa hnTXKP"
/>
`;
